examples/jwt-signer-vault/main.rs

```diff
@@ -0,0 +1,112 @@
+//! Example to sign a JWT token using Vault
+#[path = "../http/client.rs"]
+mod client;
+
+#[path = "../utils/mod.rs"]
+mod utils;
+
+use std::env;
+use std::path::Path;
+use std::str::FromStr;
+
+use base64::prelude::BASE64_STANDARD;
+use base64::Engine;
+use chrono::{TimeDelta, Utc};
+use dotenvy::dotenv;
+use jsonwebtoken::{Algorithm, Header};
+use sha2::{Digest, Sha512};
+use tracing::debug;
+use tracing_subscriber;
+use url::Url;
+use vaultrs::api::transit::requests::SignDataRequestBuilder;
+use vaultrs::api::transit::MarshalingAlgorithm;
+use vaultrs::client::{VaultClient, VaultClientSettingsBuilder};
+use vaultrs::transit::data::sign;
+
+use nr_auth::jwt::claims::Claims;
+
+/// A signed JWT should live enough for the System Identity Service to consume it.
+const DEFAULT_JWT_CLAIM_EXP: TimeDelta = TimeDelta::seconds(180);
+/// The "aud" (audience) claim identifies the recipients that the JWT is intended for.
+const DEFAULT_AUDIENCE: &str = "https://www.newrelic.com/";
+
+/// Main function to sign a JWT token with Vault
+/// It requires the following environment variables to be set:
+///
+/// VAULT_ADDRESS: Vault service address
+/// VAULT_TOKEN: Token to access Vault
+/// CLIENT_ID: Identity client_id
+/// TRANSIT_KEY_NAME: Transit Private key name
+///
+/// # Errors
+/// This function returns an error if:
+/// - The `.env` file is missing or cannot be loaded.
+/// - Required environment variables are not set.
+/// - The JWT signing process fails.
+#[tokio::main]
+async fn main() -> Result<(), Box<dyn std::error::Error>> {
+    // Set the current directory to the example's path
+    let example_dir = Path::new("examples/jwt-signer-vault");
+    env::set_current_dir(&example_dir).expect("Failed to change directory");
+
+    tracing_subscriber::fmt()
+        .with_max_level(tracing::Level::INFO)
+        .init();
+
+    dotenv()
+        .map_err(|_| ".env file not found. Copy .env.dist file to .env and fill the variables")?;
+
+    let vault_addr = utils::os::env("VAULT_ADDRESS")?;
```

Comment:
Nit: I think it would be okay to hardcode this stuff in the example.

Reply:
You might want to execute it locally without changing the example :)